### ISSUE TITLE
Setup Install...Python folder?  Cloned to where? 

### ISSUE THREAD

**Issue by holonick** [Unity-Technologies/ml-agents#1210]
Hello (again), so sorry for my noob ways here.
Is it just me or is this all a bit disjointed for everyone?
Im running into problem after problem but, no errors.
I will say, Im not a super programmer, Im new to anaconda, python ive used a little in Maya, c# Im familiar with from Unity.

**My Issues [with](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Installation.md)**
So, I am using Anaconda and have installed git to then clone the ml-agent files which I have cloned but, to where did they clone?  
Or do I need to also download the zip (github.com/Unity-Technologies/ml-agents/archive/master.zip) to then unpack to a special place for use with Unity? When I did this and using Anaconda, pointing to the unpacked files I couldnt do the "pip install ." command came back not knowing what "." is.   
Further on, when I am to "change to the python directory inside the ml-agents directory", how is this done?  
Lastly (for now), when I DID download the zip and used it in Unity there was no Python folder.  

I hope someone can tell me they went through the set up process with zero issues, so I can admire their fluidity into the ML-agents.  
You know that feeling when you are reaching for something which is just out of reach and you body begins to strain but trying to still hold its balance, thats my brain right now.  My finger tips are brushing the sides of it, just a little bit closerrrrrrrrr.

Thank you for any help you may offer.

**Comment by MarcoMeter** [comment]
Cloning the repository is equivalent to downloading the zip file.

Once you installed Anaconda and downloaded and unzipped (or cloned) this repository, you have to create your Python environment using Anaconda Prompt.

First create the conda environment:

`conda create -n ml-agents python=3.6`

After that install Tensorflow:

`pip install tensorflow==1.7.1`

The next step is to browse to the repository's directory (using cd command).

Once you got into the root directory of the repository use this command to install the remaining dependencies;

`pip install /mlagents/.`

**Comment by holonick** [comment]
**Hello again!!  Thank you so much :)**

**The conda environment step returns** "CondaValueError: prefix already exists: D:\Anaconda\envs\ml-agents"

**With the install of tensorflow==1.7.1 there is, in red,** "distributed 1.21.8 requires msgpack, which is not installed" **but, I installed it and ran it again and this message still comes up.**

**the last step of "pip install /mlagents/." returns**   "Invalid requirement: '/mlagents/.'
Traceback (most recent call last):
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\packaging\requirements.py", line 93, in __init__
    req = REQUIREMENT.parseString(requirement_string)
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 1632, in parseString
    raise exc
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 1622, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 3395, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 3717, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 3378, in parseImpl
    loc, resultlist = self.exprs[0]._parse( instring, loc, doActions, callPreParse=False )
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 1383, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\pyparsing.py", line 2689, in parseImpl
    raise ParseException(instring, loc, self.errmsg, self)
pip._vendor.pyparsing.ParseException: Expected W:(abcd...) (at char 0), (line:1, col:1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_internal\req\req_install.py", line 253, in from_line
    req = Requirement(req)
  File "d:\anaconda\envs\ml-agents\lib\site-packages\pip\_vendor\packaging\requirements.py", line 97, in __init__
    requirement_string[e.loc:e.loc + 8]))
pip._vendor.packaging.requirements.InvalidRequirement: Invalid requirement, parse error at "'/mlagent'" 


**The directory Im directing it to is D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents**

**Comment by holonick** [comment]
restarting my computer got rid of the second issue, when installing tensorflow.

the last issue of "pip install /mlagents/." still remains

**Comment by holonick** [comment]
ok, The last issue is solved.  The file "setup" was in the first "ml-agents" folder in the directory.
Thanks for your help again MarcoMeter  :)

**Comment by holonick** [comment]
But now I still dont have a "python" folder in the "ml-agents" folder

**Comment by holonick** [comment]
Am I supposed to just create an empty "python" folder?  Im watching [this ](https://www.youtube.com/watch?v=Ch2y-BSnzhk&index=2&list=LLwvxNUBEI_EBtUQ81qSUA9A&t=0s).
The folder in the video demonstration has a different structure and content to the one at [git zip](https://github.com/Unity-Technologies/ml-agents)

**Comment by dericp** [comment]
@holonick sorry for the confusion! In `v0.5` of ml-agents, we've changed some things and those video tutorials won't make sense in some places. If you want to use the video tutorials, either take a look at the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) or use an older version of ml-agents, like [0.4](https://github.com/Unity-Technologies/ml-agents/tree/0.4.0b).

**Comment by holonick** [comment]
**Hi dericp, 
Thank you!!
The confusion is major here haha.  Ill have a look at your migration link closer in a few hrs BUT, as I was trying to work it out I ran into some issues.  
Below is a cut and paste from conda to show you my path to an error.  '__main__'**

(ml-agents) C:\Users\Nick>d:

(ml-agents) D:\>python learn.py .\envs\PushBlock --train --run-id=Push-a
python: can't open file 'learn.py': [Errno 2] No such file or directory

(ml-agents) D:\>cd D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents\trainers

(ml-agents) D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents\trainers>python learn.py .\envs\PushBlock --train --run-id=Push-a
Traceback (most recent call last):
  File "learn.py", line 10, in <module>
    from .trainer_controller import TrainerController
ModuleNotFoundError: No module named '__main__.trainer_controller'; '__main__' is not a package

(ml-agents) D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents\trainers>python learn.py .\envs\PushBlock --train --run-id=Push-a
Traceback (most recent call last):
  File "learn.py", line 10, in <module>
    from .trainer_controller import TrainerController
ModuleNotFoundError: No module named '__main__.trainer_controller'; '__main__' is not a package

(ml-agents) D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents\trainers>

**Comment by MarcoMeter** [comment]
With version 0.5, which is just two days old, there is no python folder anymore and you don't use the command `python learn.py ...` anymore.

The command is now `mlagents-learn /config/config_trainer.yaml --train --run-id=test`

**Comment by holonick** [comment]
Excellent, thank you!!
By the time I get this version working, the next will come along and ill be back to square stumped again.  Hahaha.
Thanks for the help yall, Ill give that all a run soon. :)

**Comment by holonick** [comment]
Waaaait a minute.  
@MarcoMeter you say `mlagents-learn /config/config_trainer.yaml --train --run-id=test`
The migration doc says  `mlagents-learn config/trainer_config.yaml --env=3DBall --train`
BUT they both have something in common, neither worked for me...I came verrry close though haha.
Also, do I need to be navigating to the folders like I have been?  

So, firstly, with the migration doc, it tells me that my environment doesnt match.  Its there and I copy n paste the name.  Here is what it returned
```
(ml-agents) D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents>mlagents-learn config/trainer_config.yaml --env=PushBlock --train


                        ▄▄▄▓▓▓▓
                   ╓▓▓▓▓▓▓█▓▓▓▓▓
              ,▄▄▄m▀▀▀'  ,▓▓▓▀▓▓▄                           ▓▓▓  ▓▓▌
            ▄▓▓▓▀'      ▄▓▓▀  ▓▓▓      ▄▄     ▄▄ ,▄▄ ▄▄▄▄   ,▄▄ ▄▓▓▌▄ ▄▄▄    ,▄▄
          ▄▓▓▓▀        ▄▓▓▀   ▐▓▓▌     ▓▓▌   ▐▓▓ ▐▓▓▓▀▀▀▓▓▌ ▓▓▓ ▀▓▓▌▀ ^▓▓▌  ╒▓▓▌
        ▄▓▓▓▓▓▄▄▄▄▄▄▄▄▓▓▓      ▓▀      ▓▓▌   ▐▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▌   ▐▓▓▄ ▓▓▌
        ▀▓▓▓▓▀▀▀▀▀▀▀▀▀▀▓▓▄     ▓▓      ▓▓▌   ▐▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▌    ▐▓▓▐▓▓
          ^█▓▓▓        ▀▓▓▄   ▐▓▓▌     ▓▓▓▓▄▓▓▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▓▄    ▓▓▓▓`
            '▀▓▓▓▄      ^▓▓▓  ▓▓▓       └▀▀▀▀ ▀▀ ^▀▀    `▀▀ `▀▀   '▀▀    ▐▓▓▌
               ▀▀▀▀▓▄▄▄   ▓▓▓▓▓▓,                                      ▓▓▓▓▀
                   `▀█▓▓▓▓▓▓▓▓▓▌
                        ¬`▀▀▀█▓


INFO:mlagents.trainers:{'--curriculum': 'None',
 '--docker-target-name': 'None',
 '--env': 'PushBlock',
 '--help': False,
 '--keep-checkpoints': '5',
 '--lesson': '0',
 '--load': False,
 '--no-graphics': False,
 '--num-runs': '1',
 '--run-id': 'ppo',
 '--save-freq': '50000',
 '--seed': '-1',
 '--slow': False,
 '--train': True,
 '--worker-id': '0',
 '<trainer-config-path>': 'config/trainer_config.yaml'}
Process Process-1:
Traceback (most recent call last):
  File "d:\anaconda\envs\ml-agents\lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "d:\anaconda\envs\ml-agents\lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "d:\anaconda\envs\ml-agents\lib\site-packages\mlagents\trainers\learn.py", line 46, in run_training
    docker_target_name, trainer_config_path, no_graphics)
  File "d:\anaconda\envs\ml-agents\lib\site-packages\mlagents\trainers\trainer_controller.py", line 101, in __init__
    no_graphics=no_graphics)
  File "d:\anaconda\envs\ml-agents\lib\site-packages\mlagents\envs\environment.py", line 58, in __init__
    self.executable_launcher(file_name, docker_training, no_graphics)
  File "d:\anaconda\envs\ml-agents\lib\site-packages\mlagents\envs\environment.py", line 181, in executable_launcher
    .format(true_filename))
mlagents.envs.exception.UnityEnvironmentException: Couldn't launch the PushBlock environment. Provided filename does not match any environments.
```

Now, secondly, your suggested code, which is interesting that it could now functions from the Unity browser rather than from the scene build.  Heres what that returned
```
(ml-agents) D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents>mlagents-learn /config/config_trainer.yaml --train --run-id=test


                        ▄▄▄▓▓▓▓
                   ╓▓▓▓▓▓▓█▓▓▓▓▓
              ,▄▄▄m▀▀▀'  ,▓▓▓▀▓▓▄                           ▓▓▓  ▓▓▌
            ▄▓▓▓▀'      ▄▓▓▀  ▓▓▓      ▄▄     ▄▄ ,▄▄ ▄▄▄▄   ,▄▄ ▄▓▓▌▄ ▄▄▄    ,▄▄
          ▄▓▓▓▀        ▄▓▓▀   ▐▓▓▌     ▓▓▌   ▐▓▓ ▐▓▓▓▀▀▀▓▓▌ ▓▓▓ ▀▓▓▌▀ ^▓▓▌  ╒▓▓▌
        ▄▓▓▓▓▓▄▄▄▄▄▄▄▄▓▓▓      ▓▀      ▓▓▌   ▐▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▌   ▐▓▓▄ ▓▓▌
        ▀▓▓▓▓▀▀▀▀▀▀▀▀▀▀▓▓▄     ▓▓      ▓▓▌   ▐▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▌    ▐▓▓▐▓▓
          ^█▓▓▓        ▀▓▓▄   ▐▓▓▌     ▓▓▓▓▄▓▓▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▓▄    ▓▓▓▓`
            '▀▓▓▓▄      ^▓▓▓  ▓▓▓       └▀▀▀▀ ▀▀ ^▀▀    `▀▀ `▀▀   '▀▀    ▐▓▓▌
               ▀▀▀▀▓▄▄▄   ▓▓▓▓▓▓,                                      ▓▓▓▓▀
                   `▀█▓▓▓▓▓▓▓▓▓▌
                        ¬`▀▀▀█▓


INFO:mlagents.trainers:{'--curriculum': 'None',
 '--docker-target-name': 'None',
 '--env': 'None',
 '--help': False,
 '--keep-checkpoints': '5',
 '--lesson': '0',
 '--load': False,
 '--no-graphics': False,
 '--num-runs': '1',
 '--run-id': 'test',
 '--save-freq': '50000',
 '--seed': '-1',
 '--slow': False,
 '--train': True,
 '--worker-id': '0',
 '<trainer-config-path>': '/config/config_trainer.yaml'}
INFO:mlagents.envs:Start training by pressing the Play button in the Unity Editor.
INFO:mlagents.envs:
'Academy' started successfully!
Unity Academy name: Academy
        Number of Brains: 1
        Number of External Brains : 1
        Reset Parameters :

Unity brain name: PushBlockBrain
        Number of Visual Observations (per agent): 0
        Vector Observation space size (per agent): 70
        Number of stacked Vector Observation: 3
        Vector Action space type: discrete
        Vector Action space size (per agent): [7]
        Vector Action descriptions:
Process Process-1:
Traceback (most recent call last):
  File "d:\anaconda\envs\ml-agents\lib\site-packages\mlagents\trainers\trainer_controller.py", line 247, in _load_config
    with open(self.trainer_config_path) as data_file:
FileNotFoundError: [Errno 2] No such file or directory: '/config/config_trainer.yaml'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\anaconda\envs\ml-agents\lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "d:\anaconda\envs\ml-agents\lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "d:\anaconda\envs\ml-agents\lib\site-packages\mlagents\trainers\learn.py", line 47, in run_training
    tc.start_learning()
  File "d:\anaconda\envs\ml-agents\lib\site-packages\mlagents\trainers\trainer_controller.py", line 289, in start_learning
    trainer_config = self._load_config()
  File "d:\anaconda\envs\ml-agents\lib\site-packages\mlagents\trainers\trainer_controller.py", line 253, in _load_config
    .format(self.trainer_config_path))
mlagents.envs.exception.UnityEnvironmentException: Parameter file could not be found at /config/config_trainer.yaml.
```

**Comment by holonick** [comment]
@MarcoMeter **HALLELUJAH!!**
Thank you for nudging my noobiness in the right direction!!! 

`mlagents-learn D:\projects\AI_Learning\ml_Folder\ml-agents-master\config\trainer_config.yaml --env=D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents\envs\PushBlock --train`

SO excited to begin now!!

**Comment by dericp** [comment]
@MarcoMeter Thanks for helping out!

@holonick Glad you're unblocked. Let us know if you have any other problems :).

**Comment by holonick** [comment]
Thank YOU!  Super excited for this, such a game changing addition to unity.
You guys have been great! @dericp @MarcoMeter

**Comment by veerpalk** [comment]
Hi, same is happening with me @holonick  
mlagents-learn D:\projects\AI_Learning\ml_Folder\ml-agents-master\config\trainer_config.yaml --env=D:\projects\AI_Learning\ml_Folder\ml-agents-master\ml-agents\mlagents\envs\PushBlock --train

PushBlock is what ?? is it the folder uder which .exe file of unity build is kept?
and also help me with where i have to keep my build file ?

**Comment by igreenbride** [comment]
I'm having a similar issue. I was attempting to make an executable I could use with docker, but I can't seem to start the executable. I'm getting the same error where "Provided filename does not match any environments."

**Comment by harperj** [comment]
It looks like this discussion is stale / veering off topic so I'm going to close it.  With regard to running environments on Linux / Docker containers, our documentation for Docker can be found here: https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Using-Docker.md

If these instructions don't work as described feel free to open another issue.  Thanks!

**Comment by Momin504** [comment]
Unity have updated "python" folder to "ml-agents". Means in your "ml-agents" folder, you have another "ml-agents" folder which refers to "python" folder that was being used in the tutorial. Thanks!
